deployment/helm/templates/controller.yaml

```diff
@@ -36,17 +36,27 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- if .Values.controller.linuxOnly }}
-      nodeSelector:
-        "kubernetes.io/os": linux
-      {{- end }}
-      {{- if .Values.controller.onlyOnMaster }}
-      # Only run this pod on the master.
+      {{- if .Values.controller.allowOnControlPlane }}
       tolerations:
         # Allow this pod to run on the master.
         - key: node-role.kubernetes.io/master
           effect: NoSchedule
+      {{- end }}
+      {{- if or .Values.controller.linuxOnly .Values.controller.onlyOnControlPlane }}
       nodeSelector:
+        {{- if .Values.controller.nodeSelectors }}
+          {{- toYaml .Values.controller.nodeSelectors | nindent 8 }}
+        {{- end }}
+        {{- if .Values.controller.linuxOnly }}
+        "kubernetes.io/os": linux
+        {{- end }}
+        {{- if .Values.controller.onlyOnControlPlane }}
         node-role.kubernetes.io/master: ""
+        {{- end }}
+      {{- else }}
+        {{- if .Values.controller.nodeSelectors }}
+      nodeSelector:
```

Comment:
Can the nodeSelector be specified multiple times?

Reply:
it is not.  that is what this change is all about fixing.

Reply:
if is added once for `{{- if or .Values.controller.linuxOnly .Values.controller.onlyOnControlPlane }}`

and once for `{{- else }}{{- if .Values.controller.nodeSelectors }}`

Reply:
it can be simplified (the `or` can accept 3 args):

```yaml
      {{- if or .Values.controller.linuxOnly .Values.controller.onlyOnMaster .Values.controller.nodeSelectors }}
      nodeSelector:
        {{- if .Values.controller.nodeSelectors }}
          {{- toYaml .Values.controller.nodeSelectors | nindent 8 }}
        {{- end }}
        {{- if .Values.controller.linuxOnly }}
        "kubernetes.io/os": linux
        {{- end }}
        {{- if .Values.controller.onlyOnMaster }}
        node-role.kubernetes.io/master: ""
        {{- end }}
      {{- end }}
```

Reply:
Thanks, missed the else part. But agreed, we should use this simplified version.